### ISSUE TITLE
Embed Android reports an error, MissingPluginException

### ISSUE THREAD

**Issue by EightMinuteYouth** [Lyokone/flutterlocation#289]
Run directly without any problems, what happens when embedding

```
E/flutter: [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: MissingPluginException(No implementation found for method requestPermission on channel lyokone/location)
    #0      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:319)
    <asynchronous suspension>
    #1      Location.requestPermission (package:location/location.dart:70)
    #2      main.<anonymous closure> (package:user_module/main.dart:40)
    #3      _rootRunUnary (dart:async/zone.dart:1134)
    #4      _CustomZone.runUnary (dart:async/zone.dart:1031)
    #5      _FutureListener.handleValue (dart:async/future_impl.dart:139)
    #6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680)
    #7      Future._propagateToListeners (dart:async/future_impl.dart:709)
    #8      Future._completeWithValue (dart:async/future_impl.dart:524)
    #9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:32)
    #10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:290)
    #11     Application.init (package:user_module/application.dart:0)
    <asynchronous suspension>
    #12     main (package:user_module/main.dart:39)
    #13     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:45)
    #14     main (package:user_module/main.dart:20)
    #15     _runMainZoned.<anonymous closure>.<anonymous closure> (dart:ui/hooks.dart:239)
    #16     _rootRun (dart:async/zone.dart:1126)
    #17     _CustomZone.run (dart:async/zone.dart:1023)
    #18     _runZoned (dart:async/zone.dart:1518)
    #19     runZoned (dart:async/zone.dart:1502)
    #20     _runMainZoned.<anonymous closure> (dart:ui/hooks.dart:231)
    #21     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:307)
    #22     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174)
```

**Comment by Lyokone** [comment]
Hey, I'm currently updating the plugin to support Embedding V2, you can try to use master as the source in pubspec.yaml to see if it solves your problem ;)

**Comment by EightMinuteYouth** [comment]
The current version is flutter_stable_v1.12.13+hotfix.8

**Comment by Lyokone** [comment]
Please check the version 2.5.0, it should helps with embedding :)

**Comment by EightMinuteYouth** [comment]
3.5abnormal

java.lang.RuntimeException: Unable to start activity ComponentInfo{com.xx/io.flutter.embedding.android.FlutterActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.lyokone.location.FlutterLocation.setActivity(android.app.Activity)' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3304)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3443)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2040)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:224)
        at android.app.ActivityThread.main(ActivityThread.java:7520)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)

**Comment by Lyokone** [comment]
Have you migrated your project ? https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects

**Comment by Lyokone** [comment]
I've upgrade the plugin to 2.5.2 with a fix for pre-1.12 projects, please tell me if it fixes your issue :)
Feel free to reopen if it didn't
